### DISCUSSION TITLE
Picking a compiler using cross/native-file

### DISCUSSION THREAD

**Discussion by sillydan1** [mesonbuild/meson#13193] [Q&A] [answered]
Hi,

I am trying to use an avr32-bit compiler toolchain located in a local directory next to my `meson.build` file, and I am a bit confused why my approach does not work:

native-file:
```ini
# avr32.ini
[binaries]
c = 'avr32-toolchain/bin/avr32-gcc'
cpp = 'avr32-toolchain/bin/avr32-g++'
```

And then I use `meson` like so:
```sh
meson setup --native-file avr32.ini build
```

The result message seems to indicate that the compiler chosen is my system gcc, and not the one specified.
```
The Meson build system
Version: 1.4.0
Source dir: /home/agj/Documents/adventures/freertos/avr32_freertos_meson
Build dir: /home/agj/Documents/adventures/freertos/avr32_freertos_meson/build
Build type: native build
Project name: freertos_springboard
Project version: 1.0.0
C compiler for the host machine: cc (gcc 11.4.0 "cc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0")
C linker for the host machine: cc ld.bfd 2.38
Host machine cpu family: x86_64
Host machine cpu: x86_64
Build targets in project: 2

freertos_springboard 1.0.0

  User defined options
    Cross files: avr32.ini

Found ninja-1.10.1 at /usr/bin/ninja
```

Running ninja confirms this:
```sh
ninja -C build
```

```
cc <lots of flags...>
cc: error: ... does not compile ...
```

I have tried using both `--cross-file` and `--native-file` as well as manually setting `CC` and `CXX` on both the `ninja` and `meson` calls, but nothing works. I am clearly using the native/cross-file feature wrongly, but I can't figure out what I am doing wrong. Can you help?

EDIT: Versions

```
meson --version
1.4.0
ninja --version
1.10.1
```

**Comment by vbextreme** [comment, marked as answer]
The toolchain must be in the system path, otherwise, you need to use something like @GLOBAL_SOURCE_ROOT@ to refer to the current path.
add in your avr32.ini
```
[host_machine]
system     = 'baremetal'
cpu_family = 'avr32'
cpu        = 'avr32'
endian     = 'little'
```
now try 
```
$ meson --cross-file avr32.ini setup build
```

**Reply by sillydan1** [reply]
I was missing the [host_machine] part :facepalm:
